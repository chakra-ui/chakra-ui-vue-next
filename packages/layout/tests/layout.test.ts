import {
  CAspectRatio,
  CBox,
  CSquare,
  CCenter,
  CContainer,
  CCircle,
  CDivider,
  CGrid,
  CHeading,
  CLink,
  CLinkBox,
  CLinkOverlay,
  CBadge,
  CStack,
  CSimpleGrid,
  CVStack,
  CSpacer,
  CHStack,
  CGridItem,
  CKbd,
  CList,
  CListItem,
  COrderedList,
  CUnorderedList,
  CListIcon,
  CText,
  CWrap,
  CWrapItem,
  CFlex,
} from "../src"
import { render, testA11y } from "@chakra-ui/vue-test-utils"

describe("<CLink />", () => {
  const renderComponent = () =>
    render({
      components: { CLink },
      template: `<CLink>CLink</CLink>
      <CLink is-external>CLink external</CLink>
      `,
    })

  it("should render properly", () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("a11y test", async () => {
    await testA11y(renderComponent())
  })
})
describe("<CLinkBox /> <CLinkOverlay />", () => {
  const renderComponent = () =>
    render({
      components: { CHeading, CLink, CLinkBox, CLinkOverlay },
      template: `
      <c-link-box as="article">
        <c-heading as="h2" font-size="20">
          <c-link-overlay is-external href="https://vue.chakra-ui.com">Some blog post</c-link-overlay>
        </c-heading>
        <p>
          As a side note, using quotation marks around an attribute value is
          required only if this value is not a valid identifier.
        </p>
        <c-link is-external href="https://google.com">Some inner link</c-link>
      </c-link-box>
      `,
    })

  it("should render properly", () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CBadge />", () => {
  const renderComponent = () =>
    render({
      components: { CBadge },
      template: `<CBadge>this is a badge</CBadge>`,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("a11y test", async () => {
    await testA11y(renderComponent())
  })
})

describe("<CStack />", () => {
  const renderComponent = () =>
    render({
      components: { CHStack, CVStack, CStack },
      template: `
      <CStack><div>1</div><div>2</div></CStack>
      <CHStack><div>1</div><div>2</div></CHStack>
      <CVStack><div>1</div><div>2</div></CVStack>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("a11y test", async () => {
    await testA11y(renderComponent())
  })
})

describe("<CAspectRatio />", () => {
  const renderComponent = () =>
    render({
      components: { CAspectRatio },
      template: `
      <c-aspect-ratio maxW="360px" :ratio="16 / 9" mb="4">
        <img src="https://bit.ly/naruto-sage" alt="img" />
      </c-aspect-ratio>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("a11y test", async () => {
    await testA11y(renderComponent())
  })
})

describe("<CBox />", () => {
  const renderComponent = () =>
    render({
      components: { CBox },
      template: `
      <c-box bg="tomato" w="100%" p="4" color="white" max-w="300px">
        This is a div
      </c-box>
      <c-box as="section" bg="tomato" w="100%" p="4" color="white" max-w="300px">
        This is a box with as="section"
      </c-box>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CSquare /> & <CCircle />", () => {
  const renderComponent = () =>
    render({
      components: { CSquare, CCircle },
      template: `
      <c-square size="40px" bg="purple.700" color="white">
        S
      </c-square>
      <c-circle size="40px" bg="red.500" color="white">
        C
      </c-circle>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CCenter />", () => {
  const renderComponent = () =>
    render({
      components: { CCenter },
      template: `
      <c-center w="40px" h="40px" bg="purple.700" color="white">
        C
      </c-center>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CContainer />", () => {
  const renderComponent = () =>
    render({
      components: { CContainer },
      template: `
      <c-container max-w="1000px">
        container
      </c-container>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CDivider />", () => {
  const renderComponent = () =>
    render({
      components: { CDivider, CCenter },
      template: `
      <c-divider border-color="green.500" border-top-width="10px"></c-divider>
      <c-center height="50px">
        <c-divider
          border-left-width="5px"
          border-color="red.500"
          orientation="vertical"
        ></c-divider>
      </c-center>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("a11y test", async () => {
    await testA11y(renderComponent())
  })
})

describe("<CGrid /> & <CGridItem />", () => {
  const renderComponent = () =>
    render({
      components: { CBox, CGrid, CGridItem },
      template: `
      <c-grid template-columns="repeat(5, 1fr)" gap="6">
        <c-box w="100%" h="10" bg="blue.500" />
        <c-box w="100%" h="10" bg="blue.500" />
        <c-box w="100%" h="10" bg="blue.500" />
        <c-box w="100%" h="10" bg="blue.500" />
        <c-box w="100%" h="10" bg="blue.500" />
      </c-grid>
      <c-grid
        h="200px"
        templateRows="repeat(2, 1fr)"
        templateColumns="repeat(5, 1fr)"
        gap="4"
      >
        <c-grid-item rowSpan="2" colSpan="1" bg="tomato" />
        <c-grid-item colSpan="2" bg="papayawhip" />
        <c-grid-item colSpan="2" bg="papayawhip" />
        <c-grid-item colSpan="4" bg="tomato" />
      </c-grid>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CHeading />", () => {
  const renderComponent = () =>
    render({
      components: { CStack, CHeading },
      template: `
      <c-stack spacing="4">
        <c-heading as="h1" size="4xl" is-truncated>(4xl) In love with Vue & Vite & Nuxt</c-heading>
        <c-heading as="h2" size="3xl" is-truncated> (3xl) In love with Vue & Vite & Nuxt</c-heading>
        <c-heading as="h2" size="2xl">(2xl) In love with Vue & Vite & Nuxt</c-heading>
        <c-heading as="h2" size="xl"> (xl) In love with Vue & Vite & Nuxt </c-heading>
        <c-heading as="h3" size="lg"> (lg) In love with Vue & Vite & Nuxt </c-heading>
        <c-heading as="h4" size="md"> (md) In love with Vue & Vite & Nuxt </c-heading>
        <c-heading as="h5" size="sm"> (sm) In love with Vue & Vite & Nuxt </c-heading>
        <c-heading as="h6" size="xs"> (xs) In love with Vue & Vite & Nuxt </c-heading>
      </c-stack>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CKbd />", () => {
  const renderComponent = () =>
    render({
      components: { CKbd },
      template: `
      <span> <c-kbd>shift</c-kbd> + <c-kbd>H</c-kbd> </span>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CList /> <CListItem /> <CUnorderedList /> <COrderedList /> <CListIcon /> ", () => {
  const renderComponent = () =>
    render({
      components: { CList, CListItem, COrderedList, CUnorderedList, CListIcon },
      template: `
      <c-ordered-list>
        <c-list-item>Lorem ipsum dolor sit amet</c-list-item>
        <c-list-item>Facilisis in pretium nisl aliquet</c-list-item>
      </c-ordered-list>
      <c-unordered-list>
        <c-list-item>Lorem ipsum dolor sit amet</c-list-item>
        <c-list-item>Consectetur adipiscing elit</c-list-item>
      </c-unordered-list>
      <c-list>
        <c-list-item>
          <c-list-icon name="check-circle" color="green.500" />
          Lorem ipsum dolor sit amet.
        </c-list-item>
        <c-list-item>
          <c-list-icon name="exclamation" color="green.500" />
          Facilisis in pretium nisl aliquet
        </c-list-item>
      </c-list>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CSimpleGrid />", () => {
  const renderComponent = () =>
    render({
      components: { CSimpleGrid, CBox },
      template: `
      <c-simple-grid
        columns="2"
        :spacingX="['5px', '40px']"
        :spacingY="['5px', '20px']"
        minChildWidth="100px"
      >
        <c-box bg="tomato" height="80px">simple 1</c-box>
        <c-box bg="tomato" height="80px">simple 2</c-box>
        <c-box bg="tomato" height="80px">simple 3</c-box>
        <c-box bg="tomato" height="80px">simple 4</c-box>
        <c-box bg="tomato" height="80px">simple 5</c-box>
      </c-simple-grid>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CSpacer />", () => {
  const renderComponent = () =>
    render({
      components: { CFlex, CSpacer, CBox },
      template: `
      <c-flex>
        <c-box p="4" bg="red.400"> Box 1 </c-box>
        <c-spacer />
        <c-box p="4" bg="green.400"> Box 2 </c-box>
      </c-flex>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CText />", () => {
  const renderComponent = () =>
    render({
      components: { CText },
      template: `
      <c-text>normal:</c-text>
      <c-text fontSize="6xl">(6xl) In love with Vue & Vite & Nuxt</c-text>
      <c-text>truncated:</c-text>
      <c-text maxW="sm" isTruncated>
        Lorem ipsum is placeholder text commonly used in the graphic, print, and
        publishing industries for previewing layouts and visual mockups.
      </c-text>
      <c-text>line clamp:</c-text>
      <c-text :noOfLines="[1, 2, 3]" maxW="sm">
        "The quick brown fox jumps over the lazy dog" is an English-language
        pangram—a sentence that contains all of the letters of the English alphabet.
        Owing to its existence, Chakra was created.
      </c-text>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("<CWrap /> & <CWrapItem />", () => {
  const renderComponent = () =>
    render({
      components: { CWrap, CWrapItem, CCenter },
      template: `
      <c-wrap spacing="10px">
        <c-wrap-item>
          <c-center w="180px" h="80px" bg="red.200"> Box 1 </c-center>
        </c-wrap-item>
        <c-wrap-item>
          <c-center w="180px" h="80px" bg="green.200"> Box 2 </c-center>
        </c-wrap-item>
        <c-wrap-item>
          <c-center w="180px" h="80px" bg="tomato"> Box 3 </c-center>
        </c-wrap-item>
        <c-wrap-item>
          <c-center w="180px" h="80px" bg="blue.200"> Box 4 </c-center>
        </c-wrap-item>
      </c-wrap>
      `,
    })

  it("should render properly", async () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })
})
