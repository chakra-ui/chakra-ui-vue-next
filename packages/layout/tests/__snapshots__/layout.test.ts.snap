// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CAspectRatio /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-aspect-ratio chakra-w9t1xc"
  >
    <img
      alt="img"
      src="https://bit.ly/naruto-sage"
    />
  </div>
</DocumentFragment>
`;

exports[`<CBadge /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-badge chakra-1jo2llc"
  >
    this is a badge
  </div>
</DocumentFragment>
`;

exports[`<CBox /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-box chakra-2sy3sl"
  >
     This is a div 
  </div>
  <section
    class="chakra-box chakra-2sy3sl"
  >
     This is a box with as="section" 
  </section>
</DocumentFragment>
`;

exports[`<CCenter /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-center chakra-40w18w"
  >
     C 
  </div>
</DocumentFragment>
`;

exports[`<CContainer /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-container chakra-1poer1"
  >
     container 
  </div>
</DocumentFragment>
`;

exports[`<CDivider /> should render properly 1`] = `
<DocumentFragment>
  <hr
    aria-orientation="horizontal"
    class="chakra-divider chakra-1y71u2e"
  />
  <div
    class="chakra-center chakra-1yj0ggl"
  >
    <hr
      aria-orientation="vertical"
      class="chakra-divider chakra-1psxsf8"
    />
  </div>
</DocumentFragment>
`;

exports[`<CGrid /> & <CGridItem /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-grid chakra-ssvt16"
  >
    <div
      class="chakra-box chakra-62u0by"
    >
      <!---->
    </div>
    <div
      class="chakra-box chakra-62u0by"
    >
      <!---->
    </div>
    <div
      class="chakra-box chakra-62u0by"
    >
      <!---->
    </div>
    <div
      class="chakra-box chakra-62u0by"
    >
      <!---->
    </div>
    <div
      class="chakra-box chakra-62u0by"
    >
      <!---->
    </div>
  </div>
  <div
    class="chakra-grid chakra-85u0uj"
  >
    <div
      class="chakra-grid__item chakra-1q1m6d7"
    />
    <div
      class="chakra-grid__item chakra-1qvc0pb"
    />
    <div
      class="chakra-grid__item chakra-1qvc0pb"
    />
    <div
      class="chakra-grid__item chakra-1anmq07"
    />
  </div>
</DocumentFragment>
`;

exports[`<CHeading /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack chakra-17kdbol"
  >
    <h1
      class="chakra-heading chakra-cule"
    >
      (4xl) In love with Vue & Vite & Nuxt
    </h1>
    <h2
      class="chakra-heading chakra-15xpjkx"
    >
       (3xl) In love with Vue & Vite & Nuxt
    </h2>
    <h2
      class="chakra-heading chakra-1dpg06m"
    >
      (2xl) In love with Vue & Vite & Nuxt
    </h2>
    <h2
      class="chakra-heading chakra-ueokap"
    >
       (xl) In love with Vue & Vite & Nuxt 
    </h2>
    <h3
      class="chakra-heading chakra-1q5o9wg"
    >
       (lg) In love with Vue & Vite & Nuxt 
    </h3>
    <h4
      class="chakra-heading chakra-528rta"
    >
       (md) In love with Vue & Vite & Nuxt 
    </h4>
    <h5
      class="chakra-heading chakra-s5h6tf"
    >
       (sm) In love with Vue & Vite & Nuxt 
    </h5>
    <h6
      class="chakra-heading chakra-ngsgu1"
    >
       (xs) In love with Vue & Vite & Nuxt 
    </h6>
  </div>
</DocumentFragment>
`;

exports[`<CKbd /> should render properly 1`] = `
<DocumentFragment>
  <span>
    <kbd
      class="chakra-kbd chakra-o84h17"
    >
      shift
    </kbd>
     + 
    <kbd
      class="chakra-kbd chakra-o84h17"
    >
      H
    </kbd>
  </span>
</DocumentFragment>
`;

exports[`<CLink /> should render properly 1`] = `
<DocumentFragment>
  <a
    class="chakra-link chakra-8jj5m5"
  >
    CLink
  </a>
  <a
    class="chakra-link chakra-8jj5m5"
    rel="noopener noreferrer"
    target="_blank"
  >
    CLink external
  </a>
</DocumentFragment>
`;

exports[`<CLinkBox /> <CLinkOverlay /> should render properly 1`] = `
<DocumentFragment>
  <article
    class="chakra-linkbox chakra-10n1vic"
  >
    <h2
      class="chakra-heading chakra-1bhhtxl"
    >
      <a
        class="chakra-linkbox__overlay chakra-1hnz6hu"
        href="https://vue.chakra-ui.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Some blog post
      </a>
    </h2>
    <p>
       As a side note, using quotation marks around an attribute value is required only if this value is not a valid identifier. 
    </p>
    <a
      class="chakra-link chakra-8jj5m5"
      href="https://google.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Some inner link
    </a>
  </article>
</DocumentFragment>
`;

exports[`<CList /> <CListItem /> <CUnorderedList /> <COrderedList /> <CListIcon />  should render properly 1`] = `
<DocumentFragment>
  <ul
    class="chakra-list chakra-1dg6mvm"
    role="list"
  >
    <li
      class="chakra-list__item chakra-0"
    >
      Lorem ipsum dolor sit amet
    </li>
    <li
      class="chakra-list__item chakra-0"
    >
      Facilisis in pretium nisl aliquet
    </li>
  </ul>
  <ul
    class="chakra-list chakra-tu0njr"
    role="list"
  >
    <li
      class="chakra-list__item chakra-0"
    >
      Lorem ipsum dolor sit amet
    </li>
    <li
      class="chakra-list__item chakra-0"
    >
      Consectetur adipiscing elit
    </li>
  </ul>
  <ul
    class="chakra-list chakra-155za0w"
    role="list"
  >
    <li
      class="chakra-list__item chakra-0"
    >
      <svg
        class="chakra-icon chakra-vg5qi3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        
    
        <g
          stroke="currentColor"
          strokewidth="1.5"
        >
          
      
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokelinecap="round"
          />
          
      
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokelinecap="round"
          />
          
      
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokemiterlimit="10"
          />
          
    
        </g>
        
  
      </svg>
       Lorem ipsum dolor sit amet. 
    </li>
    <li
      class="chakra-list__item chakra-0"
    >
      <svg
        class="chakra-icon chakra-vg5qi3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        
    
        <g
          stroke="currentColor"
          strokewidth="1.5"
        >
          
      
          <path
            d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
            fill="none"
            strokelinecap="round"
          />
          
      
          <path
            d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
            fill="currentColor"
            strokelinecap="round"
          />
          
      
          <circle
            cx="12"
            cy="12"
            fill="none"
            r="11.25"
            strokemiterlimit="10"
          />
          
    
        </g>
        
  
      </svg>
       Facilisis in pretium nisl aliquet 
    </li>
  </ul>
</DocumentFragment>
`;

exports[`<CSimpleGrid /> should render properly 1`] = `
<DocumentFragment>
  <ul
    class="chakra-simple-grid chakra-12tmpmw"
  >
    <div
      class="chakra-box chakra-9rloc5"
    >
      simple 1
    </div>
    <div
      class="chakra-box chakra-9rloc5"
    >
      simple 2
    </div>
    <div
      class="chakra-box chakra-9rloc5"
    >
      simple 3
    </div>
    <div
      class="chakra-box chakra-9rloc5"
    >
      simple 4
    </div>
    <div
      class="chakra-box chakra-9rloc5"
    >
      simple 5
    </div>
  </ul>
</DocumentFragment>
`;

exports[`<CSpacer /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-flex chakra-1xhj18k"
  >
    <div
      class="chakra-box chakra-zzdd9b"
    >
       Box 1 
    </div>
    <div
      class="chakra-spacer chakra-17xejub"
    />
    <div
      class="chakra-box chakra-pp1cg2"
    >
       Box 2 
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CSquare /> & <CCircle /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-square chakra-yh3vrj"
  >
     S 
  </div>
  <div
    class="chakra-circle chakra-1afsfyy"
  >
     C 
  </div>
</DocumentFragment>
`;

exports[`<CStack /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack chakra-n21gh5"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
  <div
    class="chakra-stack-horizontal chakra-x9juev"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
  <div
    class="chakra-stack-vertical chakra-n21gh5"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
  </div>
</DocumentFragment>
`;

exports[`<CText /> should render properly 1`] = `
<DocumentFragment>
  <p
    class="chakra-text chakra-0"
  >
    normal:
  </p>
  <p
    class="chakra-text chakra-n2bes8"
  >
    (6xl) In love with Vue & Vite & Nuxt
  </p>
  <p
    class="chakra-text chakra-0"
  >
    truncated:
  </p>
  <p
    class="chakra-text chakra-3urtkt"
  >
     Lorem ipsum is placeholder text commonly used in the graphic, print, and publishing industries for previewing layouts and visual mockups. 
  </p>
  <p
    class="chakra-text chakra-0"
  >
    line clamp:
  </p>
  <p
    class="chakra-text chakra-3jijpv"
  >
     "The quick brown fox jumps over the lazy dog" is an English-language pangramâ€”a sentence that contains all of the letters of the English alphabet. Owing to its existence, Chakra was created. 
  </p>
</DocumentFragment>
`;

exports[`<CWrap /> & <CWrapItem /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="chakra-wrap chakra-0"
  >
    <ul
      class="chakra-wrap__list chakra-gbdy1k"
    >
      <li
        class="chakra-wrap__listItem chakra-1yp4ln"
      >
        <div
          class="chakra-center chakra-58a5xs"
        >
           Box 1 
        </div>
      </li>
      <li
        class="chakra-wrap__listItem chakra-1yp4ln"
      >
        <div
          class="chakra-center chakra-1cf3c28"
        >
           Box 2 
        </div>
      </li>
      <li
        class="chakra-wrap__listItem chakra-1yp4ln"
      >
        <div
          class="chakra-center chakra-1un2hvj"
        >
           Box 3 
        </div>
      </li>
      <li
        class="chakra-wrap__listItem chakra-1yp4ln"
      >
        <div
          class="chakra-center chakra-dd3w5r"
        >
           Box 4 
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
